src/Services/Upgrade.php

```diff
@@ -68,6 +78,10 @@ protected function lastModifiedAt($upgrade): Carbon
 
     private function canRun($upgrade): bool
     {
+        if ($upgrade instanceof PreMigration ^ $this->preMigration) {
```

Comment:
what is this operator?  ("^")

Reply:
XOR, if the upgrade is premigration and preMigration is false then we shouldn't run it and if the upgrade isn't preMigration and preMigration is true we shouldn't run it.

Reply:
and a xor b is equal to  (!a & b) or (a & !b)

Reply:
👍🏻